core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -185,6 +187,10 @@
    * time. */
   private static final ReentrantLock LOCK = new ReentrantLock();
 
+  static {
+    System.setProperty("calcite.enable.regenerate.metadata.handler", "false");
```

Comment:
it looks like you have to clear this system property in a finally block at the end of the test and you also have to set it only at the beginning of the test

Reply:
Thanks for the feedback.

It seems to make the setting take effect, we have to set the system property before class CalciteSystemProperty is loaded. So we have to set it in the static constructor.